paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FullCacheLookupTable.java

```diff
@@ -91,7 +110,20 @@ public FullCacheLookupTable(Context context) {
             this.userDefinedSeqComparator = null;
             this.appendUdsFieldNumber = 0;
         }
+
+        Options options = Options.fromMap(context.table.options());
         this.projectedType = projectedType;
+        this.refreshAsync = options.get(LOOKUP_REFRESH_ASYNC);
+        this.refreshExecutor =
+                this.refreshAsync
+                        ? Executors.newSingleThreadExecutor(
+                                new ExecutorThreadFactory(
+                                        String.format(
+                                                "%s-lookup-refresh",
+                                                Thread.currentThread().getName())))
+                        : null;
```

Comment:
We can use `MoreExecutors.newDirectExecutorService()`, which can unify the code about `refreshExecutor` without having to judge whether it is `null` or `refreshAsync`.